src/Pact/Typechecker.hs

```diff
@@ -994,11 +994,12 @@ toUserType t = case t of
   _ -> die (_tInfo t) $ "toUserType: expected user type: " ++ show t
   where
     derefUT (Ref r) = toUserType' (fmap Left r :: Term (Either Ref n))
-    derefUT Direct {} = die (_tInfo t) $ "toUserType: unexpected direct ref: " ++ show t
+    derefUT (Direct d) = toUserType' (fmap Right d)
+    -- derefUT Direct {} = die (_tInfo t) $ "toUserType: unexpected direct ref: " ++ show t
```

Comment:
this commented line is unreachable now. Do you want to keep it?

Reply:
naw, will fix